app/lib/generate_consent_documents.rb

```diff
@@ -0,0 +1,37 @@
+class GenerateConsentDocuments
+  def create_consent_docs
+    # GenerateConsentDocuments.new.create_consent_docs
+    intakes_without_consent_doc = []
+    successes = []
+    problems = []
+
+    Intake.all.each do |intake|
+      consent_form_exists = false
+
+      intake.documents.each do |doc|
+        if doc.document_type == DocumentTypes::Form14446.key
+          consent_form_exists = true
+        end
+      end
+
+      intakes_without_consent_doc << intake.id unless consent_form_exists || intake.primary_consented_to_service_at.nil?
```

Comment:
If you can use the `.primary_consented_to_service_at` column to decrease the number of Intake records you loop across, that would also provide a substantial speed-up. Not a blocker.